scipy/spatial/transform/_rotation.pyx

```diff
@@ -223,6 +230,131 @@ cdef double[:, :] _compute_euler_from_matrix(
 
     return angles
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef double[:, :] _compute_euler_from_quat(
+    np.ndarray[double, ndim=2] quat, const uchar[:] seq, bint extrinsic=False
+):
+    # The algorithm assumes intrinsic frame transformations. The algorithm
+    # in the paper is formulated for rotation matrices which are transposition
+    # rotation matrices used within Rotation.
+    # Adapt the algorithm for our case by
+    # 1. Instead of transposing our representation, use the transpose of the
+    #    O matrix as defined in the paper, and be careful to swap indices
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+    
+    if not extrinsic:
+        seq = seq[::-1]
+
+    cdef int i = _elementary_basis_index(seq[0])
+    cdef int j = _elementary_basis_index(seq[1])
+    cdef int k = _elementary_basis_index(seq[2])
+
+    is_proper = i == k
+    if is_proper:
+        k = 3 - i - j # get third axis
+        
+    # Step 0
+    # Check if permutation is even (+1) or odd (-1)     
+    cdef int sign = int((i-j)*(j-k)*(k-i)/2)
```

Comment:
I believe an integer division is used here and the combination is equal to +2 or -2?

Int conversion shouldn't be necessary here (my guess).

Reply:
The expression `(i-j)*(j-k)*(k-i)` is always equal to -2 or 2, and then I divide it by 2 (this is a way to calculate the parity of the permutation IJK).

The int conversion is indeed probably not needed.

Reply:
Good trick to know how to determine parity.

Reply:
It is indeed!

Reply:
Just tested removing the `int` conversion, cython throws a compile error!